### PR TITLE
Custom model data

### PR DESCRIPTION
**Pull request by MBDOFF**
Adding the posibility to add custom model data to items.

### CODE REVIEW

File: plugin/src/main/java/com/iridium/iridiumcore/Item.java [hunk 1]
```diff
@@ -93,6 +94,27 @@ public Item(XMaterial material, int slot, int amount, String displayName, String
...
+    /**
+     * Creates a new item with the provided data.
+     * Used for creating player heads.
+     *
+     * @param material    The material of the item. Specifies the type
+     * @param slot        The slot where this item should be in
+     * @param amount      The amount of this item. Should not be higher than the max stack size
+     * @param displayName The display name of the item with color codes
+     * @param headOwner   The owner of the head
+     * @param lore        The lore of the item, can be empty
```
Comment (MertUnverdi) [inline plugin/src/main/java/com/iridium/iridiumcore/Item.java:106]: Add a param for model data tho

File: plugin/src/main/java/com/iridium/iridiumcore/Item.java [hunk 2]
```diff
@@ -93,6 +94,27 @@ public Item(XMaterial material, int slot, int amount, String displayName, String
...
+     */
+    public Item(XMaterial material, int slot, int amount, String displayName, String headOwner, List<String> lore, Integer model) {
```
Comment (MertUnverdi) [inline plugin/src/main/java/com/iridium/iridiumcore/Item.java:108]: U need add some other constructors for items without headdata or slot params etc.

**Review by MertUnverdi — APPROVED** [review]

**Review by MertUnverdi — APPROVED** [review]

**Comment by PeachesMLG** [comment]
With this many constructors, it would be better to implement a builder pattern (with a few default constructors for legacy support)

**Comment by dlsf** [comment]
I agree with Peaches, it's really getting out of hand now.

But I'm also wondering: How would you use that custom model data option? What kind of integer is that?

**Comment by MBDOFF** [comment]
I optimized the model data constructors.

**Review by PeachesMLG — APPROVED** [review]

**Comment by Fantom250YT** [comment]
Any reason this isn't merged?

File: build.gradle.kts [hunk 3]
```diff
@@ -5,7 +5,7 @@ plugins {
 }
 
 group = "com.iridium"
-version = "1.3.9"
+version = "1.4.0"
```
Comment (DoctaEnkoda) [inline build.gradle.kts:8]: Need rebase

**Review by DoctaEnkoda — CHANGES_REQUESTED** [review]

**Pull request closed without merging**